Audio support for WCL

ujfalusi opened the pull request:
PCI ID and support for Wildcat Lake audio

ujfalusi committed:
PCI: pci_ids: add INTEL_HDA_WCL

Add PCI id for Wildcat Lake audio.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi committed:
ASoC: SOF: Intel: add initial support for WCL

Clone PTL and adjust the number of cores from 5 to 3.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi committed:
ALSA: hda: intel-dsp-config: Add WCL support

WCL uses the same receipt as PTL and PTL-H

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi committed:
ALSA: hda: hda-intel: add Wildcat Lake support

One more PCI ID.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

bardliao commented on include/linux/pci_ids.h:
```diff
@@ -3050,6 +3050,7 @@
 #define PCI_DEVICE_ID_INTEL_HDA_EHL_0	0x4b55
 #define PCI_DEVICE_ID_INTEL_HDA_EHL_3	0x4b58
 #define PCI_DEVICE_ID_INTEL_HDA_JSL_N	0x4dc8
+#define PCI_DEVICE_ID_INTEL_HDA_WCL	0x4d28
```
Would it be better to reorder with ID incremental? I.e.
```c
#define PCI_DEVICE_ID_INTEL_HDA_EHL_3	0x4b58
#define PCI_DEVICE_ID_INTEL_HDA_WCL	0x4d28
#define PCI_DEVICE_ID_INTEL_HDA_JSL_N	0x4dc8
```

ujfalusi replied:
Yes, it should be

ujfalusi commented on sound/soc/sof/intel/pci-ptl.c:
```diff
@@ -55,10 +55,40 @@ static const struct sof_dev_desc ptl_desc = {
+	.default_fw_filename = {
+		[SOF_IPC_TYPE_4] = "sof-wcl.ri",
+	},
+	.nocodec_tplg_filename = "sof-wcl-nocodec.tplg",
```
How should this be?
we use the `snd_soc_acpi_intel_ptl_machines` and `snd_soc_acpi_intel_ptl_sdw_machines` but the nocodec is using `sof-wcl-`
Should we duplicate the ptl mach with s/ptl/wcl applied or change the nocodec to sof-ptl as well?

WCL is in essence PTL with reduced number of cores.

kv2019i replied:
I think in general WCL tplgs should be separate. Let's have nocodec separate as well. sof-ptl-nocodec.tplg only uses cores 0+1+2 now, but if this is changed at some point, it'll break WCL machines using this (albeit this is for R&D but still).
UPDATE: I'll change my mind, ptl-nocodec is ok.

ujfalusi commented:
Changes since v1:
- correct the order of PCI IDs
- Use sof-ptl-nocodec as default nocodec topology name for now as the ptl match tables are used as well.

kv2019i reviewed (approved):
After some back and forth, I'm now ok with this in its current form.

ranj063 commented on sound/soc/sof/intel/ptl.c:
```diff
@@ -126,6 +126,29 @@ const struct sof_intel_dsp_desc ptl_chip_info = {
+	.power_down_dsp = mtl_power_down_dsp,
+	.disable_interrupts = lnl_dsp_disable_interrupts,
+	.hw_ip_version = SOF_INTEL_ACE_3_0,
+};
```
@ujfalusi this patch is good for now but yuou know what I'm thinking. We should abstract some of the fields here that are largely unmodified gen to gen. something to consider doing in the future.
